### PR TITLE
feat: add a placeholder `README` file in codegen'd directory

### PR DESCRIPTION
**Pull request by kanadgupta**
| 🚥 Resolves RM-8188 |
| :------------------- |

## 🧰 Changes

This PR enhances our SDK generation to automatically create placeholder `README.md` files in the codegen'd directories.

As part of this, I made a few test enhancements so the `UPDATE_FIXTURES` env variable will now create any missing fixtures.

## 🧬 QA & Testing

Does the copy look good? And do tests pass?

### CODE REVIEW

File: packages/api/src/codegen/languages/typescript/index.ts [hunk 1]
```diff
@@ -497,6 +499,32 @@ sdk.server('https://eu.api.example.com/v14');`),
...
+This SDK was autogenerated by the [\`api\` SDK generator](https://api.readme.dev), powered by [ReadMe](https://readme.com) 🦉
+
+Add SDK setup information and usage examples here so your users get started in a jiffy! 🚀
+
+<!---
+
+Here's some additional info about the generated SDK:
+
+\`api\` version: ${PACKAGE_VERSION}
+Generated at ${new Date().toISOString()}
```
Comment (erunion) [inline packages/api/src/codegen/languages/typescript/index.ts:518]: After https://github.com/readmeio/api/pull/775 is merged in you should be able to grab the `createdAt` date from the lockfile for this with this code:

```ts
storage.getFromLockfile().createdAt
```

File: packages/api/test/codegen/languages/typescript/index.test.ts [hunk 2]
```diff
@@ -124,6 +140,14 @@ describe('typescript', () => {
   });
 
   describe('#compile', () => {
+    beforeEach(() => {
+      vi.setSystemTime(new Date('2023'));
```
Comment (erunion) [inline packages/api/test/codegen/languages/typescript/index.test.ts:144]: Was about to ask to do some sort of `<<package version>>` thing for the date in the README then noticed this.

pretty-sneaky-bro-sneaky

**Review by erunion — APPROVED** [review]

File: packages/api/src/codegen/languages/typescript/index.ts (hunk 1 above)
Comment (erunion) [reply packages/api/src/codegen/languages/typescript/index.ts:518]: should be able to pull this in now that `createdAt` has been added to the lockfile.

File: packages/api/src/codegen/languages/typescript/index.ts [hunk 3]
```diff
@@ -497,6 +499,35 @@ sdk.server('https://eu.api.example.com/v14');`),
     return sourceFile;
   }
 
+  /**
+   * Create a placeholder `README.md` file in the repository, with information on how to use/administer the SDK.
+   *
+   */
+  createREADME() {
+    let createdAt = new Date().toISOString();
+    const currentAPI = Storage.getLockfile().apis.find(api => api.identifier === this.identifier);
```
[the remark this answers is not in the document]
Comment (kanadgupta) [reply packages/api/src/codegen/languages/typescript/index.ts:508]: yeah i noticed that `getFromLockfile` method but wasn't able to get it working without some deeper refactors 😮‍💨

> Maybe the whole storage class should just be refactored to be entirely static. 🤔

into this idea!

**Pull request merged**